java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -1276,11 +1279,16 @@ private static Table sliceUpColumnVectors(SerializedTableHeader header,
       long[] nullCounts = header.nullCounts;
       long numRows = header.getNumRows();
       int numColumns = dataTypes.length;
-      ColumnVector[] vectors = new ColumnVector[numColumns];
+      ColumnVector[] vectors = new ColumnVector[Math.max(1,numColumns)];
       DeviceMemoryBuffer validity = null;
       DeviceMemoryBuffer data = null;
       DeviceMemoryBuffer offsets = null;
       try {
+        if (numColumns == 0) {
+          vectors[0] = new ColumnVector(DType.INT32, numRows, Optional.of(0L),
+              DeviceMemoryBuffer.allocate(DType.INT32.sizeInBytes * numRows), validity, offsets);
```

Comment:
Do we really want to allocate `numRows` of INT32 values when they're not going to be used?  Ideally we shouldn't be allocating anything, including a libcudf `table`, in this scenario.  If we _really_ have to allocate something then it should be a `BOOL8` column.

Reply:
I tried out passing nulls for buffers and non zero rows and it does not return the right values. I will again look into this as I agree allocation is totally synthetic and was not my first choice.